### PR TITLE
[Hexagon] Fix concat lowering for HVX for 64B vector length

### PR DESCRIPTION
**Pull request by quic-santdas**
When concatenation of vector instructions is formed, as a part of it vector rotation is performed. The direction of the shift was not correctly calculated. This fixes the rotation factor.

### CODE REVIEW

**Comment by quic-santdas** [comment]
@iajbar Please review.

**Review by quic-santdas — COMMENTED** [review]
The extra function attributes which were not needed for the testcase have been removed.

**Review by iajbar — APPROVED** [review]
LGTM

File: llvm/lib/Target/Hexagon/HexagonISelLoweringHVX.cpp [hunk 1]
```diff
@@ -1748,7 +1748,7 @@ HexagonTargetLowering::LowerHvxConcatVectors(SDValue Op, SelectionDAG &DAG)
 
   unsigned InpLen = ty(Op.getOperand(0)).getVectorNumElements();
   MVT ByteTy = MVT::getVectorVT(MVT::i8, HwLen);
-  SDValue S = DAG.getConstant(InpLen*BitBytes, dl, MVT::i32);
```
[the remark this answers is not in the document]
Comment (quic-santdas) [reply llvm/lib/Target/Hexagon/HexagonISelLoweringHVX.cpp:1751]: Thanks for your comment! Yes, it does in the testcase. The shufflevectors are lowered to vror instructions for rotation [v0 = vror(v1,r1)]. The rotation is calculated correctly now (r1 in this case). We can see the difference in generated assembly after this fix.

**Comment by quic-santdas** [comment]
Ping for the way forward. Should this be merged?

**Review by androm3da — APPROVED** [review]

**Review by SundeepKushwaha — APPROVED** [review]
lgtm

**Pull request merged**